widgets/btc_linux.lua

```diff
@@ -0,0 +1,57 @@
+---------------------------------------------------
+-- Licensed under the GNU General Public License v2
+--  * (c) 2017, 0x5b <dragen15051@gmail.com>
+---------------------------------------------------
+
+-- {{{ Grab environment
+local setmetatable = setmetatable
+local pcall = pcall
+local helpers = require("vicious.helpers")
+local https = require("socket.http")
+local spawn = require("awful.spawn")
+local json = require("json")
+local string = {
+    sub = string.sub,
+    upper = string.upper,
+}
+-- }}}
+
+
+-- Btc: provides current bitcoin price
+-- vicious.widgets.btc
+local btc_linux = {}
+
+
+-- {{ Bitcoin widget type
+function btc_linux.async(format, warg, callback)
+    -- Default values
+    if not warg then warg = "usd" end
+
+    local btc = { ["{price}"] = "N/A" }
+    local currency_code = string.upper(warg)
+    local url = "https://api.coindesk.com/v1/bpi/currentprice/"..helpers.escape(currency_code)..".json"
+    local cmd = "curl --connect-timeout 1 "..url
```

Comment:
the function name `escape` here is missleading and not the right function for shell escaping.
Use shellquote instead:

```diff
-    local url = "https://api.coindesk.com/v1/bpi/currentprice/"..helpers.escape(currency_code)..".json"
-    local cmd = "curl --connect-timeout 1 "..url
+    local url = "https://api.coindesk.com/v1/bpi/currentprice/" .. currency_code .. ".json"
+    local cmd = "curl "..helpers.shellquote(url)
```

`connect-timeout` is also not necessary, when the call is asynchronous.